### DISCUSSION TITLE
(De)Duplication of nested environments

### DISCUSSION THREAD

**Discussion by liquidcarbon** [prefix-dev/pixi#583] [Ideas]
There's a pattern I use with exploratory work in notebooks, where I'd have a root `pixi.toml` and then folders with their own `pixi.toml`.

It's similar to how examples are set up here, or to how one would have a base conda and more custom stuff on top.

Is there a way / plans for Pixi to become smart around not installing the same dependencies into nested environments?  Especially python itself.  How about in the event there is the same python version in the parent folder, the inner `.pixi` folder creates symlinks to the outer python binaries, but then installs its own packages?

In my workflows, I've got the first cell of Jupyter notebooks (which runs from the base pixi layer) doing the opposite - adding the inner layer to `sys.path`:

```
import sys
local_packages = sys.path[0] + "/.pixi/env/lib/python3.12/site-packages"
sys.path.append(local_packages)
sys.path

['/home/coder/kr/posts/p231219_read_npy',
 '/home/coder/kr/.pixi/env/lib/python312.zip',
 '/home/coder/kr/.pixi/env/lib/python3.12',
 '/home/coder/kr/.pixi/env/lib/python3.12/lib-dynload',
 '',
 '/home/coder/kr/.pixi/env/lib/python3.12/site-packages',
 '/home/coder/kr/posts/p231219_read_npy/.pixi/env/lib/python3.12/site-packages']
```

Both pixi folders contain ~20000 files that are likely 99% identical.

What can we do?

**Comment by baszalmstra** [comment]
Pixi already hardlinks files from a central cache. This means that unless you delete the cache in between runs the disk space is most likely already deduplicated between _all_ environments on your machine.

**Reply by wolfv** [reply]
If you want to test that, you can look at the `inode` of the file. They should all be identical :)

**Reply by liquidcarbon** [reply]
Oh that's lovely!
